allennlp/data/dataset_readers/reading_comprehension/triviaqa.py

```diff
@@ -0,0 +1,171 @@
+        tarball.
+    tokenizer : ``Tokenizer``, optional
+        We'll use this tokenizer on questions and evidence passages, defaulting to
+        ``WordTokenizer`` if none is provided.
+    token_indexers : ``Dict[str, TokenIndexer]``, optional
+        Determines how both the question and the evidence passages are represented as arrays.  See
+        :class:`TokenIndexer`.  Default is to have a single word ID for every token.
+    """
+    def __init__(self,
+                 base_tarball_path: str,
+                 unfiltered_tarball_path: str = None,
+                 tokenizer: Tokenizer = None,
+                 token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        self._base_tarball_path = base_tarball_path
+        self._unfiltered_tarball_path = unfiltered_tarball_path
+        self._tokenizer = tokenizer or WordTokenizer()
+        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
+
+    @overrides
+    def read(self, file_path: str):
+        logger.info("Opening base tarball file at %s", self._base_tarball_path)
+        base_tarball = tarfile.open(cached_path(self._base_tarball_path), 'r')
+        if 'unfiltered' in file_path:
+            logger.info("Opening unfiltered tarball file at %s", self._unfiltered_tarball_path)
+            unfiltered_tarball = tarfile.open(cached_path(self._unfiltered_tarball_path), 'r')
+            logger.info("Loading question file from tarball")
+            data_json = json.loads(unfiltered_tarball.extractfile(file_path).read().decode('utf-8'))
+        else:
+            logger.info("Loading question file from tarball")
+            path = os.path.join('qa', file_path)
+            data_json = json.loads(base_tarball.extractfile(path).read().decode('utf-8'))
+
+        logger.info("Reading the dataset")
+        instances = []
+        for question_json in tqdm(data_json['Data']):
+            question_text = question_json['Question']
+            question_tokens = self._tokenizer.tokenize(question_text)
+
+            evidence_files: List[List[str]] = []  # contains lines from each evidence file
+            if 'web' in file_path:
+                for result in question_json['SearchResults']:
+                    filename = result['Filename']
+                    evidence_file = base_tarball.extractfile(os.path.join("evidence", "web", filename))
+                    evidence_files.append([line.decode('utf-8') for line in evidence_file.readlines()])
+            else:
+                for result in question_json['EntityPages']:
+                    filename = result['Filename']
+                    evidence_file = base_tarball.extractfile(os.path.join("evidence", "wikipedia", filename))
+                    evidence_files.append([line.decode('utf-8') for line in evidence_file.readlines()])
+
+            answer_json = question_json['Answer']
+            answer_texts = answer_json['NormalizedAliases'] + answer_json.get('HumanAnswers', [])
```

Comment:
The human answers are stored in the normalized form. The official evaluation script runs `normalize_answer` on them before evaluation. To avoid mixing un-normalized and normalized answers I would pre-normalize these answers as well.

Reply:
Thanks, good point.